test/stream.js

```diff
@@ -1,61 +1,15 @@
 import {Buffer} from 'node:buffer';
-import {exec} from 'node:child_process';
 import {once} from 'node:events';
 import process from 'node:process';
 import {setTimeout} from 'node:timers/promises';
-import {promisify} from 'node:util';
 import test from 'ava';
 import getStream from 'get-stream';
-import {pEvent} from 'p-event';
 import {execa, execaSync} from '../index.js';
-import {setFixtureDir, FIXTURES_DIR} from './helpers/fixtures-dir.js';
-
-const pExec = promisify(exec);
+import {setFixtureDir} from './helpers/fixtures-dir.js';
+import {fullStdio, getStdio} from './helpers/stdio.js';
 
 setFixtureDir();
 
-const checkEncoding = async (t, encoding) => {
-	const {stdout} = await execa('noop-no-newline.js', [STRING_TO_ENCODE], {encoding});
-	t.is(stdout, BUFFER_TO_ENCODE.toString(encoding));
-
-	const {stdout: nativeStdout} = await pExec(`node noop-no-newline.js ${STRING_TO_ENCODE}`, {encoding, cwd: FIXTURES_DIR});
-	t.is(stdout, nativeStdout);
-};
-
-// This string gives different outputs with each encoding type
-const STRING_TO_ENCODE = '\u1000.';
-const BUFFER_TO_ENCODE = Buffer.from(STRING_TO_ENCODE);
-
-test('can pass encoding "utf8"', checkEncoding, 'utf8');
-test('can pass encoding "utf-8"', checkEncoding, 'utf8');
-test('can pass encoding "utf16le"', checkEncoding, 'utf16le');
-test('can pass encoding "utf-16le"', checkEncoding, 'utf16le');
-test('can pass encoding "ucs2"', checkEncoding, 'utf16le');
-test('can pass encoding "ucs-2"', checkEncoding, 'utf16le');
-test('can pass encoding "latin1"', checkEncoding, 'latin1');
-test('can pass encoding "binary"', checkEncoding, 'latin1');
```

Comment:
Those tests were incorrect. For example, `binary` should be `binary` not `latin1`. And `ucs2` should be `ucs2` not `utf16le`. Technically those values are the same, but the point of the test is to try each alternative name of each encoding.